handler-proxy/src/main/java/io/netty/handler/proxy/ProxyHandler.java

```diff
@@ -287,7 +287,7 @@ private void setConnectSuccess() {
             connectTimeoutFuture.cancel(false);
```

Comment:
should we do `connectTimeoutFuture = null;` here (and other spots where we call `connectTimeoutFuture.cancel(..);`)?